server.js

```diff
@@ -5728,6 +5728,62 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Opera add-ons integration
+camp.route(/^\/opera-add-ons\/(v|d|rating|stars|rating-count)\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  const type = match[1];
+  const name = match[2];
+  const format = match[3];
+  const badgeData = getBadgeData('opera add-ons', data);
+  const url = 'https://addons.opera.com/en/extensions/details/' + name;
+  request(url, function(err, res, buffer) {
+    if (!err) {
+      try {
+        const cheerio = require('cheerio');
+        const $ = cheerio.load(buffer);
+        const version = $('section.about > dl > dd').eq(2).text();
+        const downloads = parseInt($('section.about > dl > dd').eq(0).text().replace(/,/g, ''));
+        const ratingValue = parseFloat($('meta[itemprop="ratingValue"]').eq(0).attr('content'));
+        const reviewCount = parseInt($('span[itemprop="reviewCount"]').eq(0).text().replace(/,/g, ''));
+        const vdata = versionColor(version);
+        switch (type) {
+        case 'v':
+          badgeData.text[1] = vdata.version;
+          badgeData.colorscheme = vdata.color;
+          break;
+        case 'd':
+          badgeData.text[0] = data.label || 'downloads';
+          badgeData.text[1] = metric(downloads) + ' total';
+          badgeData.colorscheme = downloadCountColor(downloads);
+          break;
+        case 'rating':
+          badgeData.text[0] = data.label || 'rating';
+          badgeData.text[1] = ratingValue.toFixed(2) + '/5';
+          badgeData.colorscheme = floorCountColor(ratingValue, 2, 3, 4);
+          break;
+        case 'stars':
+          badgeData.text[0] = data.label || 'rating';
+          badgeData.text[1] = starRating(Math.round(ratingValue));
+          badgeData.colorscheme = floorCountColor(ratingValue, 2, 3, 4);
+          break;
+        case 'rating-count':
+          badgeData.text[0] = data.label || 'rating-count';
+          badgeData.text[1] = metric(reviewCount) + ' total';
+          badgeData.colorscheme = floorCountColor(reviewCount, 5, 50, 500);
```

Comment:
A lot of this code looks like it was copied and pasted from the Chrome web store implementation. Instead of copying and pasting it, could you pull it out into a reusable utility function?